saas_demo/models/saas_operator.py

```diff
@@ -78,23 +80,32 @@ def update_addons_path(self):
     @api.multi
     def restart_odoo(self):
         if self.is_local():
+            test = self.env['ir.config_parameter'].get_param('test_saas_demo')
+            if test:
+                # no need to restart odoo folder in test mode
+                return
             service.server.restart()
 
     @api.multi
     def _update_repos(self):
         self.ensure_one()
         if self.type != 'local':
             return
-        return self.demo_id.repo_ids._local_update_repo()
-
-    @api.multi
-    def update_ad_paths(self, path):
-        if self.is_local():
-            if path not in ad_paths:
-                ad_paths.append(path)
-
-    @api.multi
-    def remove_ad_paths(self, path):
-        if self.is_local():
-            if path in ad_paths:
-                ad_paths.remove(path)
+        has_updates = False
+        for repo in self.demo_id.repo_ids:
+            updated = self.server_update_repo(repo.url, repo.url_escaped, repo.branch, repo.commit)
+            if updated:
+                has_updates = True
+        return has_updates
+
+    @staticmethod
+    def server_update_repo(url, url_escaped, branch, commit):
+        repos_root = repos_dir()
+        updated = False
+        repos_path = os.path.join(repos_root, branch, url_escaped)
+        if not os.path.isdir(os.path.join(repos_path)):
+            updated = True
+        current_commit = update_repo(repos_path, url, branch)
+        if current_commit != commit:
+            updated = True
+        return updated
```

Comment:
Why it doesn't check type != local ?

Reply:
Check of this condition is performed in the method above. It's a static method

Reply:
Use ``local`` in method name. Also, you may need to make method private (i.e. start name with underscore)

Reply:
Done

Reply:
Why don't you make it private?